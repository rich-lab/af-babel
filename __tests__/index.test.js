const { resolve } = require('path')
const afBabel = require('../lib')

const cwd = resolve(__dirname, '..')
const removeCwd = path => path.replace(cwd, '')

const expectConfigToMatchSnapshot = config => {
  const { plugins = [], presets = [], ...rest } = config
  const normalize = item => {
    if (Array.isArray(item)) {
      return [removeCwd(item[0]), ...item.slice(1)]
    } else if (typeof item === 'string') {
      return removeCwd(item)
    }
  }

  config = {
    plugins: plugins.map(normalize),
    presets: presets.map(normalize),
    ...rest,
  }

  expect(config).toMatchSnapshot()
}

it('default config', function () {
  const babel = afBabel()
  expectConfigToMatchSnapshot(babel)
})

it('presetOptions', function () {
  const babel = afBabel({
    presetOptions: {
      env: {
        targets: {
          chrome: 73
        },
      }
    }
  })
  expectConfigToMatchSnapshot(babel)
})

it('babelConfig', function () {
  const babel = afBabel({
    babelConfig: {
      plugins: [
        ['babel-plugin-import', { libraryName: 'antd-mobile', style: true }]
      ]
    }
  })
  expectConfigToMatchSnapshot(babel)
})
