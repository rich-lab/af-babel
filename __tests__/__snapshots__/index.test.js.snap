// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelConfig 1`] = `
Object {
  "babelrc": false,
  "plugins": Array [
    Array [
      "babel-plugin-import",
      Object {
        "libraryName": "antd-mobile",
        "style": true,
      },
    ],
  ],
  "presets": Array [
    Array [
      "/node_modules/_@umijs_babel-preset-umi@3.0.0-beta.22@@umijs/babel-preset-umi/app.js",
      Object {
        "import": Array [
          Object {
            "libraryName": "antd-mobile",
            "style": true,
          },
          Object {
            "camel2DashComponentName": false,
            "libraryDirectory": "",
            "libraryName": "lodash",
          },
        ],
        "nodeEnv": "production",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`default config 1`] = `
Object {
  "babelrc": false,
  "plugins": Array [],
  "presets": Array [
    Array [
      "/node_modules/_@umijs_babel-preset-umi@3.0.0-beta.22@@umijs/babel-preset-umi/app.js",
      Object {
        "import": Array [
          Object {
            "libraryName": "antd-mobile",
            "style": true,
          },
          Object {
            "camel2DashComponentName": false,
            "libraryDirectory": "",
            "libraryName": "lodash",
          },
        ],
        "nodeEnv": "production",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`presetOptions 1`] = `
Object {
  "babelrc": false,
  "plugins": Array [],
  "presets": Array [
    Array [
      "/node_modules/_@umijs_babel-preset-umi@3.0.0-beta.22@@umijs/babel-preset-umi/app.js",
      Object {
        "env": Object {
          "targets": Object {
            "chrome": 73,
          },
        },
        "import": Array [
          Object {
            "libraryName": "antd-mobile",
            "style": true,
          },
          Object {
            "camel2DashComponentName": false,
            "libraryDirectory": "",
            "libraryName": "lodash",
          },
        ],
        "nodeEnv": "production",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;
